test/odbc/psqlodbc/psqlodbc_tests_common.h

```diff
@@ -0,0 +1,696 @@
+    comma = ",";
+  }
+  tableConstraints += ")";
+
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(tableName, tableColumns, tableConstraints));
+  odbcHandler.CloseStmt();
+
+  vector<tuple<int, int, SQLPOINTER, int>> constraints_bind_columns = {
+    {1, SQL_C_CHAR, columnName, CHARSIZE}
+  };
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(constraints_bind_columns));
+
+  const string UNIQUE_QUERY = 
+    "SELECT C.COLUMN_NAME FROM "
+    "INFORMATION_SCHEMA.TABLE_CONSTRAINTS T "
+    "JOIN INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE C "
+    "ON C.CONSTRAINT_NAME=T.CONSTRAINT_NAME "
+    "WHERE "
+    "C.TABLE_NAME='" + tableName.substr(tableName.find('.') + 1, tableName.length()) + "' "
+    "AND T.CONSTRAINT_TYPE='UNIQUE'";
+  
+  odbcHandler.ExecQuery(UNIQUE_QUERY);
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  EXPECT_EQ(rcode, SQL_SUCCESS);
+  EXPECT_EQ(string(columnName), uniqueConstraintColumns[0]);
+  odbcHandler.CloseStmt();
+
+  const int BUFFER_LENGTH = 8192;
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN *>> bind_columns = {
+    {1, SQL_C_LONG, &pk, 0, &pk_len},
+    {2, type, &data, bufferLen, &data_len}
+  };
+
+  string insertString{};
+  comma = "";
+  for (int i = 0; i < insertedValues.size(); i++) {
+      insertString += comma + "(" + std::to_string(i) + "," + insertedValues[i] + ")";
+      comma = ",";
+  }
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(tableName, insertString));
+  
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affectedRows);
+  EXPECT_EQ(rcode, SQL_SUCCESS);
+  EXPECT_EQ(affectedRows, insertedValues.size());
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(tableName, {"*"}, vector<string> {orderByColumnName}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < insertedValues.size(); i++) {
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    EXPECT_EQ(rcode, SQL_SUCCESS);
+    EXPECT_EQ(pk_len, 4);
+    EXPECT_EQ(pk, i);
+    if (insertedValues[i] != "NULL") {
+      EXPECT_EQ(data_len, expectedLen[i]);
+      EXPECT_EQ(data, expectedInsertedValues[i]);
+    }
+    else {
+      EXPECT_EQ(data_len, SQL_NULL_DATA);
+    }
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  EXPECT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", tableName));
+}
+
+template <typename T>
+void testComparisonFunctionsNumeric(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, int type, vector<T> colResults, int bufferLen, vector<string> insertedValues, vector<string> operationsQuery, vector<T> expectedResults, vector<long> expectedLen) {
```

Comment:
It seems like the logic for testComparisonFunctionsNumeric and testArithmeticOperators is the same (or almost?)

In fact the functions do not care (or should not) what the operations are, as long as they can be used in select expression, as in:
`  odbcHandler.ExecQuery(SelectStatement(tableName, operationsQuery, vector<string>{}));`

Thus, it is probably possible to have one function named generically, e.g. 
**testOperations**

Similarly, there are corresponding versions for characters (testComparisonFunctionsChar). Perhaps also characters could be handled by the same generic function?

Reply:
They are slightly different. In `testArithmeticOperators` the expected results is a 2D array (`const vector<vector<T>> &expectedResults`) while in `testComparisonFunctions` it's a 1D array (`const vector<T> &expectedResults`).

The set up in these functions look the same, but in the assert & verify step they look different.